src/core/RecyclerListView.tsx

```diff
@@ -391,7 +410,15 @@ export default class RecyclerListView<P extends RecyclerListViewProps, S extends
         }
         if (forceFullRender || this.props.layoutProvider !== newProps.layoutProvider || this.props.isHorizontal !== newProps.isHorizontal) {
             //TODO:Talha use old layout manager
-            this._virtualRenderer.setLayoutManager(newProps.layoutProvider.newLayoutManager(this._layout, newProps.isHorizontal));
+            if (forceFullRender) {
```

Comment:
This isn't what we want

Reply:
If layoutProvider and orientation changes we do the old thing. If only forceFullRender we use cached layouts.

Reply:
Moved forceFullRender to separate if condition